appinventor/components/src/com/google/appinventor/components/runtime/Menu.java

```diff
@@ -0,0 +1,266 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2019 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.runtime;
+
+import android.app.Activity;
+import android.util.Log;
+import android.view.MenuItem.OnMenuItemClickListener;
+import com.google.appinventor.components.annotations.DesignerComponent;
+import com.google.appinventor.components.annotations.DesignerProperty;
+import com.google.appinventor.components.annotations.PropertyCategory;
+import com.google.appinventor.components.annotations.SimpleEvent;
+import com.google.appinventor.components.annotations.SimpleObject;
+import com.google.appinventor.components.annotations.SimpleProperty;
+import com.google.appinventor.components.common.ComponentCategory;
+import com.google.appinventor.components.common.PropertyTypeConstants;
+import com.google.appinventor.components.common.YaVersion;
+import com.google.appinventor.components.runtime.util.YailList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@DesignerComponent(version = YaVersion.MENU_COMPONENT_VERSION,
```

Comment:
Add a Javadoc to describe more fully what this class does and how it is used. This is used to create the user facing documentation so you'll want it to be easily understood by non-programmers. Once you build the system, it will update docs/markdown/components/layout.md and docs/html/components/layout.html based on what you put here.